pkg/backends/backend.go

```diff
@@ -0,0 +1,11 @@
+package backends
+
+func DeleteBackend(backends []string, backendUrl string) ([]string, bool) {
```

Comment:
This screams "use a map to store state!" Maps in golang have constant time performance, arrays have linear time performance at best.

Reply:
If you mean changing the type for whole sprayproxy backends as map. Then we will have time performance issue before we start sprayproxy itself. I can see we are adding backends as space separated array [here](https://github.com/redhat-appstudio/infra-deployments/blob/multi-cluster/components/sprayproxy/add-backends.yml) which means after we read [viper backend value](https://github.com/redhat-appstudio/sprayproxy/blob/main/cmd/server.go#L36), will convert it into map (which will be linear).

Another place is where we actually started running the server I see we are printing [message](https://github.com/redhat-appstudio/sprayproxy/blob/main/pkg/server/server.go#L60) which is comma separated string from a slice.

IIUC then this change will look like [commit](https://github.com/redhat-appstudio/sprayproxy/pull/28/commits/618e3c782b0661d819f3750e76177393af86c72f) and with this approach I was confused with values of map, so I prefered to keep it empty.